force-app/main/default/classes/SOQL.cls

```diff
@@ -85,9 +108,19 @@ public with sharing class SOQL {
      * @return A map of Task Ids to Task records.
      * Update the corresponding test class and testGetTasksDueNextWeek method to cover this method
      */
-    //[public | private ] [static] data_type method_name (input parameters) {
+    public static Map<Id, Task> getTasksDueNext7Days () {
     //  The body of the method
-    //}
+    Map<Id, Task> result = new Map<Id, Task>();
+    Date eightDaysFromNow = Date.today().addDays(8);
+    List<Task> tasks = [SELECT Id
+                        FROM Task
+                        WHERE ActivityDate >= TODAY 
+                        AND ActivityDate <= :eightDaysFromNow];
```

Comment:
Take a look at some of the special key words you can use in queries like TODAY, YESTERDAY, LAST_N_DAYS:n

These can simplify your code and remove the need to instantiate dates in some (but defiinitely not all) situations.